### PR TITLE
Shift live stream manifest data forward so that it can play back

### PR DESCRIPTION
**Pull request by jw-kaurand**
During a live stream, align equivalent segments in a newly loaded manifest with the most recent one loaded.  This allows us to get relatively accurate start and end times for the segments so that we can play back the segments even if we have a discontinuity or other anomaly in between.

JW7-3199

### CODE REVIEW

File: src/controller/stream-controller.js [hunk 1]
```diff
@@ -811,7 +811,37 @@ class StreamController extends EventHandler {
         duration = newDetails.totalduration,
         sliding = 0;
 
-    logger.log(`level ${newLevelId} loaded [${newDetails.startSN},${newDetails.endSN}],duration:${duration}`);
+    console.log(`level ${newLevelId} loaded [${newDetails.startSN},${newDetails.endSN}],duration:${duration}`);
```
Comment (robwalch) [inline src/controller/stream-controller.js]: Shouldn't you continue to use the logger?

File: src/controller/stream-controller.js [hunk 2]
```diff
@@ -811,7 +811,37 @@ class StreamController extends EventHandler {
...
+            oldSegment = oldFrags[sameFragIndex],
+            diffEstimate = oldSegment.start - newSegment.start;
+        logger.log(`new quality level has equivalent fragments. shifting playlist over ${diffEstimate} seconds to align`);
+        for (newFragIndex = 0; newFragIndex < newFrags.length; newFragIndex++) {
+          newFrags[newFragIndex].start += diffEstimate;
+          newFrags[newFragIndex].end += diffEstimate;
+        }
+      } else {
+        this.levelLastLoaded = newLevelId;
+        logger.error(`fatal error - new quality level ${newLevelId} does not have equivalent fragments (sn values) with old quality level ${this.levelLastLoaded}`);
```
Comment (robwalch) [inline src/controller/stream-controller.js]: sn = Sequence Number

Comment (robwalch) [inline src/controller/stream-controller.js]: Is the issue sequence number or discontinuity? Does the PTS match.

**Comment by jw-kaurand** [comment]
A different solution for this using discontinuity sequence ids is more appropriate

**Comment by haydenm315** [comment]
I've encountered a similar problem, and your solution helped me.  I play a static loading stream while waiting for a live stream to generate ts segments.  Once enough segments are created, I use discontinuity and switch to the live stream.  The live stream has presentation timestamps which start again at 0.  

I ended up adjusting startPTS and endPTS in mergeDetails which exists in level-helper.  There is a conditional which returns if end < start.  I simply take the endPTS from the old details fragments, and append this to all the fragments in the new details.

**Pull request closed without merging**